### PR TITLE
IDEMPIERE 4702 Messagebox enhancement: Esc key closes Dialog with cancel action

### PR DESCRIPTION
**Pull request by Soundbytes**
https://idempiere.atlassian.net/browse/IDEMPIERE-4702

### CODE REVIEW

File: org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/component/Messagebox.java [hunk 1]
```diff
@@ -55,6 +57,8 @@
 	 * 
 	 */
 	private static final long serialVersionUID = 8928526331932742124L;
+	
+	private static final CLogger log = CLogger.getCLogger(Messagebox.class);
```
Comment (CarlosRuiz-globalqss) [inline org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/component/Messagebox.java:61]: This line is creating an eclipse warning - if not used is better to remove it

File: org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/component/Messagebox.java [hunk 2]
```diff
@@ -387,15 +399,19 @@ else if (event.getTarget() == btnIgnore)
 		{
 			returnValue = IGNORE;
 		}
-
+		close();
+	}
+	
+	private void close() {
+		SessionManager.getSessionApplication().getKeylistener().removeEventListener(Events.ON_CTRL_KEY, this);
```
Comment (CarlosRuiz-globalqss) [inline org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/component/Messagebox.java:406]: Just a question - why is this line required?  Is zk adding the listener internally when you add the annotation?

**Review by CarlosRuiz-globalqss — CHANGES_REQUESTED** [review]
Hi @Soundbytes, the pull request must be done against a branch with name IDEMPIERE-4702.

If you create the pull request over master it can become a problem very soon to resolve conflicts and merging.

I tested this successfully, can you please do the suggested change?

Regards,

Carlos Ruiz

**Comment by CarlosRuiz-globalqss** [comment]
Also, in the commit message please add the link IDEMPIERE-4702  (with dash) so JIRA can connect the pull request.

File: org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/component/Messagebox.java (hunk 1 above)
Comment (Soundbytes) [reply org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/component/Messagebox.java:61]: That was left over from debugging - will be removed

File: org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/component/Messagebox.java (hunk 2 above)
Comment (Soundbytes) [reply org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/component/Messagebox.java:406]: Oops. I first tried a different approach using a the KeyListener. That worked equally well but was a bit more complicated so I removed it. Except for this line which I just had forgotten. - will be removed

**Comment by Soundbytes** [comment]
> Hi @Soundbytes, the pull request must be done against a branch with name IDEMPIERE-4702.

It is not possible to change the source branch after the PR is publlished. I am replacing this request with a new one.
